### ISSUE TITLE
`/var/task/node_modules/openai/_shims/agent-node.mjs` import issue in T3 stack with Vercel Edge function

### ISSUE THREAD

**Issue by jacoblee93** [openai/openai-node#297]
### Confirm this is a Node library issue and not an underlying OpenAI API issue

- [X] This is an issue with the Node library

### Describe the bug

The SDK throws an import error with the default T3 stack config (https://create.t3.gg/) in Edge functions with the `pages` router. 

First reported in LangChainJS, reproed by me with the SDK by itself: https://github.com/hwchase17/langchainjs/issues/2558

Here is the trace:

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/node_modules/openai/_shims/agent-node.mjs' imported from /var/task/node_modules/openai/core.mjs
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:329:11)
    at moduleResolve (node:internal/modules/esm/resolve:992:10)
    at moduleResolveWithNodePath (node:internal/modules/esm/resolve:936:12)
    at defaultResolve (node:internal/modules/esm/resolve:1178:79)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
RequestId: 8a11b070-7a2c-4a19-8bdd-16db8dce3f14 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

### To Reproduce

1. Set up an app using the T3 Stack: https://create.t3.gg/
2. Set up an endpoint under `pages/api/route.ts`
3. Import and use the OpenAI SDK (code below)
4. Deploy to Vercel
5. Ping the endpoint

### Code snippets

```JavaScript
import type { NextApiRequest, NextApiResponse } from 'next'
import { type ClientOptions, OpenAI as OpenAIClient } from "openai";

type ResponseData = {
  message: string
}

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<ResponseData>
) {
  const client = new OpenAIClient();
  const result = await client.chat.completions.create({
    messages: [{
      role: "user",
      content: "hi",
    }],
    model: "gpt-3.5-turbo",
  })
  res.status(200).json({ message: JSON.stringify(result) })
}
```

### OS

Vercel Edge function

### Node version

Edge

### Library version

openai v4.5.0

**Comment by rattrayalex** [comment]
Hey @jacoblee93 thanks for reporting! Sorry about this, we're working on it. 

Closing as a dup of https://github.com/openai/openai-node/issues/296

**Comment by jacoblee93** [comment]
This is what I tried: https://github.com/jacoblee93/my-t3-app

Then I deployed to Vercel with the CLI.

Only difference I see at a glance is that you used `pnpm` or `npm` and I used `yarn`?

**Comment by jacoblee93** [comment]
Ohh actually it looks like I'm not using the edge runtime?

**Comment by rattrayalex** [comment]
Hmm, I tried with and without that line, but I'll try your repo. Did you deploy to Vercel with `vercel deploy`?

**Comment by rattrayalex** [comment]
Strangely enough, I'm able to repro by moving to `yarn`. Still scratching my head on why that might be, but it's a start. Thanks very much, Jacob!

**Comment by jacoblee93** [comment]
Yeah pretty odd - we unblocked by just making our `openai` dep a tilde on `4.4.0` (which is probably better practice anyway).

Let me know if I can do anything else.

**Comment by vladern** [comment]
Yes, on version 4.4.0 seems to be working fine.

**Comment by rattrayalex** [comment]
This was fixed in openai `4.6.0`, so you should be able to upgrade to that as well.
